dubbo-remoting/dubbo-remoting-http12/src/main/java/org/apache/dubbo/remoting/http12/netty4/h1/NettyHttp1ConnectionHandler.java

```diff
@@ -0,0 +1,88 @@
+import org.apache.dubbo.remoting.http12.h1.Http1ServerTransportListenerFactory;
+import org.apache.dubbo.remoting.http12.message.JsonCodec;
+import org.apache.dubbo.rpc.model.FrameworkModel;
+
+import java.util.concurrent.Executor;
+
+public class NettyHttp1ConnectionHandler extends SimpleChannelInboundHandler<Http1Request> {
+
+    private Http1ServerTransportListenerFactory http1ServerTransportListenerFactory;
+
+    private Http1ServerTransportListener http1TransportListener;
+
+    private final FrameworkModel frameworkModel;
+
+    private final URL url;
+
+    private final Executor executor;
+
+    private Http1ServerChannelObserver errorResponseObserver;
+
+    public NettyHttp1ConnectionHandler(URL url, FrameworkModel frameworkModel) {
+        this.url = url;
+        this.frameworkModel = frameworkModel;
+        this.executor = url.getOrDefaultFrameworkModel().getExtensionLoader(ThreadPool.class).getAdaptiveExtension().getExecutor(url);
+    }
+
+    public NettyHttp1ConnectionHandler(URL url,
+                                       FrameworkModel frameworkModel,
+                                       Http1ServerTransportListenerFactory http1ServerTransportListenerFactory) {
+        this.url = url;
+        this.frameworkModel = frameworkModel;
+        this.executor = url.getOrDefaultFrameworkModel().getExtensionLoader(ThreadPool.class).getAdaptiveExtension().getExecutor(url);
+        this.http1ServerTransportListenerFactory = http1ServerTransportListenerFactory;
+    }
+
+    public void setHttp1ServerTransportListenerFactory(Http1ServerTransportListenerFactory http1ServerTransportListenerFactory) {
+        this.http1ServerTransportListenerFactory = http1ServerTransportListenerFactory;
+    }
+
+    protected void channelRead0(ChannelHandlerContext ctx, Http1Request http1Request) {
+        //process h1 request
+        initTransportListenerIfNecessary(ctx);
+        executor.execute(() -> {
+            try {
+                http1TransportListener.onMetadata(http1Request);
+                http1TransportListener.onData(http1Request);
+            } catch (Exception e) {
+                errorResponseObserver.onError(e);
+            }
+        });
+    }
+
+    private void initTransportListenerIfNecessary(ChannelHandlerContext ctx) {
+        if (http1TransportListener == null) {
+            Http1ServerTransportListenerFactory http1ServerTransportListenerFactory = this.http1ServerTransportListenerFactory;
+            Assert.notNull(http1ServerTransportListenerFactory, "http1ServerTransportListenerFactory must be not null.");
+            http1TransportListener = http1ServerTransportListenerFactory.newInstance(new NettyHttp1Channel(ctx.channel()), url, frameworkModel);
+        }
+        this.errorResponseObserver = new Http1ServerChannelObserver(new NettyHttp1Channel(ctx.channel()));
+        this.errorResponseObserver.setHttpMessageCodec(JsonCodec.INSTANCE);
```

Comment:
The json should not be written dead here

Reply:
ok, i will redesign here